ospd_openvas/daemon.py

```diff
@@ -247,6 +253,52 @@ def format_vt_modification_time(self, value: str) -> str:
 
         return datetime.utcfromtimestamp(int(value)).strftime("%Y%m%d%H%M%S")
 
+    def get_filtered_vts_list(self, vts, vt_filter: str) -> Optional[List[str]]:
+        """ Gets a collection of vulnerability test from the redis cache,
+        which match the filter.
+
+        Arguments:
+            vt_filter: Filter to apply to the vts collection.
+            vts: The complete vts collection.
+
+        Returns:
+            List with filtered vulnerability tests. The list can be empty.
+            None in case of filter parse failure.
+        """
+        if not vt_filter:
+            raise OspdCommandError('vt_filter: A valid filter is required.')
+
+        filters = self.parse_filters(vt_filter)
+        if not filters:
+            return None
+
+        if not self.nvti:
+            return None
+
+        vt_oid_list = [vtlist[1] for vtlist in self.nvti.get_oids()]
+        vt_oid_list_temp = copy.copy(vt_oid_list)
+        vthelper = VtHelper(self.nvti)
+
+        for _element, _oper, _filter_val in filters:
```

Comment:
Just a hint. There is no need to use underscore names here. The underscore names in loops should only indicate that this variable isn't used within the for loop (to make the linters happy too). E.g.

```python
for _id, value in some_list:
    do_something_with_the_value(value)
```

Additional info. In classes underscores are used to indicate private methods or variables of the class that could be accessed by other code technically but shouldn't.

```python
class Foo:
  def __init__(self):
    self._private_var = 'foo'

  def _private_method(self):
    self.do_something()
```